src/main/java/gregtech/api/metatileentity/MetaTileEntityHolder.java

```diff
@@ -128,7 +128,10 @@ public boolean hasCapability(@Nonnull Capability<?> capability, @Nullable EnumFa
     public void update() {
         if (metaTileEntity != null) {
             metaTileEntity.update();
+        } else if (world.isRemote) { // Remove null TEs on the client
+            this.getWorld().setBlockToAir(this.getPos());
         }
+        
```

Comment:
This code is simple, but it makes a big impact. I wonder how you test it? 
- How to ensure data synch between the server and the client? 
- How you test it in low tps that you mentioned?
- `It appears that they can return to normal MTEs if the server syncs its data properly.` How is this confirmed?

I think the core problem is to solve why MTE is synced to empty, but `BlockMachine` not.

Reply:
1. I have noticed several times that if this accidentally hits, the server seems to be able to recover it; besides, I would think that TE updates on the server go to the client, which could recover the block itself. I've also noticed an occasional flicker sometimes in low TPS while placing blocks, but it always properly recovers.
 2. I put a Thread.sleep of 1 in MetaTileEntity's update, which quite easily makes a nice configurable lag.
 3. Again, it also looks like placing is made a little weird in low TPS environments, and I can see the MTE flicker for a tick, and then return, as if the client code deleted the block, but the server returned it.

Reply:
As you said, I simulated a low TPS environment (4). But didnt recovering, with a completely invisible block blocking me.  
![worse2](https://user-images.githubusercontent.com/18493855/144573690-a4d06a42-c37d-4b7f-9134-8b2570d8d943.gif)
.
The good news is that I seem to have figured out how to reproduce invisible blocks inspired by you and have some ideas.